src/codegen/internal-module-registry-scanner.ts

```diff
@@ -7,6 +7,7 @@ export function createInternalModuleRegistry(basedir: string) {
     .flatMap(dir => readdirRecursive(path.join(basedir, dir)))
     .filter(file => file.endsWith(".js") || (file.endsWith(".ts") && !file.endsWith(".d.ts")))
```

Comment:
☝️ There's a helper internal to Node's path that would simplify the above (gets the longest ext first). Nice to see a case for it externally.